trademapper/js/trademapper.mapper.js

```diff
@@ -108,6 +104,116 @@ define(["d3", "topojson", "worldmap", "disputedareas", "countrycentre", "config"
 				.attr("fill", "url(#diagonalHatch)");
 	},
 
+	// load overlays from the overlays= variable in the query string
+	addOverlays: function () {
+		var params = (new URL(document.location)).searchParams;
+		var urls = params.get('overlays');
+
+		if (urls === null) {
+			return;
+		}
+
+		urls = urls.split(',');
+		console.log('FETCHING OVERLAYS FROM URLS:', urls);
+
+		urls.forEach(this.loadTopojsonFromURL.bind(this));
+	},
+
+	loadTopojsonFromURL: function(url) {
+		var self = this;
+
+		d3.json(url).then(
+			this.loadTopojson.bind(this),
+
+			function (error) {
+				console.log("unable to download", error);
+			}
+		);
+	},
+
+	loadTopojson: function (data) {
+		// lines need drawing with class overlay-line;
+		// polygons need drawing with class overlay-area, but also need to
+		// add lines denoting their boundaries;
+		// points can be treated the same way as polygons and don't need boundaries
+		// to be derived
+		var polygons = {
+			type: "GeometryCollection",
+			geometries: []
+		};
+
+		var lines = {
+			type: "GeometryCollection",
+			geometries: []
+		};
+
+		var points = {
+			type: "GeometryCollection",
+			geometries: []
+		};
+
+		// sort the geometries so they can be drawn appropriately
+		var geometries, i, geometry;
+		for (var propertyName in data.objects) {
+			geometries = data.objects[propertyName].geometries;
```

Comment:
Are we happy with this just failing silently if the data is not formatted as expected? I don't know if Trademapper already has a system for reporting import errors to the user?

Reply:
I'll add some error catching.